packages/esm-patient-medications-app/src/order-basket/medication-order-form.component.tsx

```diff
@@ -221,55 +210,19 @@ export default function MedicationOrderForm({ initialOrderBasketItem, onSign, on
             <>
               <Grid className={styles.gridRow}>
                 <Column md={4}>
-                  <ComboBox
+                  <NumberInput
                     id="doseSelection"
                     light={isTablet}
-                    items={dosageOptions}
-                    selectedItem={
-                      dosageOptions?.length
-                        ? {
-                            id: orderBasketItem.dosage ? `${orderBasketItem.dosage?.value}` : null,
-                            text: orderBasketItem.dosage ? `${orderBasketItem.dosage?.value}` : null,
-                          }
-                        : null
-                    }
-                    // @ts-ignore
                     placeholder={t('editDoseComboBoxPlaceholder', 'Dose')}
-                    titleText={t('editDoseComboBoxTitle', 'Enter Dose')}
-                    itemToString={(item) => item?.text}
-                    onChange={({ selectedItem }) => {
-                      if (selectedItem) {
-                        selectedItem.id = selectedItem.id == 'draft' ? selectedItem.text : selectedItem.text;
-                        setOrderBasketItem({
-                          ...orderBasketItem,
-                          dosage: { value: Number(selectedItem.text) },
-                        });
-                      } else {
-                        setOrderBasketItem({
-                          ...orderBasketItem,
-                          dosage: { value: 0 },
-                        });
-                      }
-                      // cleaup
-                      setDosageOptions(dosageOptions.filter((opt) => opt.id != 'draft'));
-                    }}
-                    onInputChange={(value) => {
-                      if (value == 'undefined') {
-                        return;
-                      }
-                      const valueExists = value ? dosageOptions.some((opt) => `${opt.text}` == value) : null;
-                      const draftIndex = dosageOptions.findIndex((opt) => opt.id == 'draft');
-                      // validate and clean up
-                      if (draftIndex >= 0 && value) {
-                        dosageOptions[draftIndex].text = value;
-                      } else if (draftIndex >= 0) {
-                        dosageOptions.pop();
-                      }
-                      if (value && !valueExists && draftIndex == -1 && !isNaN(Number(value))) {
-                        dosageOptions.push({ id: 'draft', text: value });
-                      }
+                    label={t('editDoseComboBoxTitle', 'Enter Dose')}
+                    value={orderBasketItem?.dosage ?? 0}
+                    onChange={(e, { value }) => {
+                      setOrderBasketItem({
+                        ...orderBasketItem,
+                        dosage: value ? parseFloat(value) : 0,
```

Comment:
Why not `parseInt`?

Reply:
Technically, the dosage could be a float, unfortunately.

Reply:
Ok